test/functional/api/v2/organization_controller_test.rb

```diff
@@ -0,0 +1,243 @@
+  end
+
+  test "should get organizations for nested object" do
+    @organization.domain_ids = [domains(:mydomain).id]
+    get :index, {:domain_id => domains(:mydomain).to_param }
+    assert_response :success
+    assert_equal assigns(:organizations), [taxonomies(:organization1)]
+  end
+
+  #####################
+  # test config/initializers/rabl_init.rb
+  # using Organization as class to test rabl extension
+  test "root name on index should be results by default" do
+    get :index, {}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    assert response['results'].is_a?(Array)
+    refute response['organizations']
+  end
+
+  test "root name on index is configured to be controller name" do
+    Rabl.configuration.use_controller_name_as_json_root = true
+    get :index, {}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    refute response['results']
+    assert response['organizations'].is_a?(Array)
+  end
+
+  test "root name on index can be overwritten by param root_name" do
+    get :index, {:root_name => "data"}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    assert response['data'].is_a?(Array)
+    refute response['results']
+    refute response['organizations']
+  end
+
+  test "on index no object_root name for each element in array" do
+    get :index, {}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    assert response['results'].is_a?(Array)
+    assert_equal ['created_at', 'id', 'name', 'title', 'updated_at'], response['results'][0].keys.sort
+  end
+
+  test "object name on show defaults to object class name" do
+    obj = taxonomies(:organization1)
+    get :show, {:id => obj.id}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    klass_name = obj.class.name.downcase
+    assert "organization", klass_name
+    assert response.is_a?(Hash)
+    assert_equal obj.id, response["id"]
+  end
+
+  test "object name on show can be specified" do
+    obj = taxonomies(:organization1)
+    get :show, {:id => obj.id, :root_name => 'row'}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    assert response['row'].is_a?(Hash)
+    assert_equal obj.id, response['row']["id"]
+  end
+
+  test "no object name on show" do
+    obj = taxonomies(:organization1)
+    get :show, {:id => obj.id, :root_name => 'false'}
+    response = ActiveSupport::JSON.decode(@response.body)
+    assert response.is_a?(Hash)
+    assert_equal obj.id, response["id"]
+  end
+
+  # using Organization as class to pagination and search metadata
+  # create 26 organizations per name per letter A-Z
+  def add_organizations
+    Array('a'..'z').each do |letter|
+      Organization.create(:name => letter)
+    end
+  end
+
+  test "should return correct metadata if no params passed" do
```

Comment:
Could you wrap this in a context `metadata` so that you make a setup block to `add_organizations`? Is creating 29 organizations absolutely necessary? It's going to hit the db really heavily (read: slow tests), please look into how to mock these results, or work with expectations for metadata.